srcs/juloo.keyboard2/Emoji.java

```diff
@@ -7,81 +7,782 @@
 import java.io.BufferedReader;
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 
 public class Emoji
 {
-  private final String _name;
   private final KeyValue _kv;
-  private final String _desc;
 
-  private static HashMap<String, Emoji> emojis_by_name = new HashMap<String, Emoji>();
-
-  protected Emoji(String name, String bytecode, String desc)
+  protected Emoji(String bytecode)
   {
-    _name = name;
-    _kv = new KeyValue(bytecode, KeyValue.Kind.String, 0, 0);
-    _desc = desc;
-    emojis_by_name.put(name, this);
-  }
-
-  public String name()
-  {
-    return _name;
+    this._kv = new KeyValue(bytecode, KeyValue.Kind.String, 0, 0);
   }
 
   public KeyValue kv()
   {
     return _kv;
   }
 
-  public String getDescription()
+
+  private final static List<Emoji> _all = new ArrayList<>();
+  private final static List<List<Emoji>> _groups = new ArrayList<>();
+  private final static HashMap<String, Emoji> _stringMap = new HashMap<>();
+
+  public static void init(Resources res)
   {
-    return (_desc);
-  }
+    if (!_all.isEmpty())
+      return;
+
+    try
+    {
+      InputStream inputStream = res.openRawResource(R.raw.emojis);
+      BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
+      String line;
+
+      // Read emoji (until empty line)
+      while (!(line = reader.readLine()).isEmpty())
+      {
+        Emoji e = new Emoji(line);
+        _all.add(e);
+        _stringMap.put(line, e);
+      }
 
-  public static int num_groups = 0;
+      // Read group indices
```

Comment:
This format works but I think could be slightly simpler with the group indices on the first line. That way, the list don't have to be delimited.
The code that interprets the indices can still be done after the entire list has been read.